### DISCUSSION TITLE
How can i set dynamic name for file for recording video using app created by API

### DISCUSSION THREAD

**Discussion by Nikita-Goncharov** [OvenMediaLabs/OvenMediaEngine#1737] [Help]
## Hello, i need to create dynamic name of file(with some id or datetime) for recording video, but my app on ovenmediaengine was created with internal API, request body example:
```
"name": "cam-0",
  "type": "live",
  "outputProfiles": {
      "outputProfile": [
          {
              "name": "stream",
              "outputStreamName": "stream_output",
              "encodes": {
                  "videos": [
                      {
                          "name": "bypass_video",
                          "bypass": True
                      }
                  ]
              }
          }
      ]
  },
  "providers": {
      "ovt": {},
      "rtmp": {},
      "rtspPull": {},
      "srt": {},
      "webrtc": {}
  },
  "publishers": {
      "llhls": {},
      "ovt": {},
      "webrtc": {},
      "file": {
          "rootPath": "/home/name/videos",
          "filePath": "video_record.mp4"
      }
  }
}
```
I saw in docs that i can do like this in static xml config:
```
<FilePath>/path/to/${VirtualHost}/${Application}/${Stream}/
              ${StartTime:YYYYMMDDhhmmss}_${Sequence}.mp4</FilePath>
```
But i need to do same thing with datetime or with some unique id using OME API.
Is there a way how i can do it ?

**Comment by Keukhan** [comment]
You can start and stop recording using the REST API. Although this is not documented in the GitBook, please refer to the sample below.
Simply set the paths for filePath and infoPath. When these paths are set through the REST API, the settings in Server.xml won’t be used.

```
{
    "id": "user_define_id",
    "stream": {
        "name": "stream"
        // "variantNames": ["h264", "aac"]
    },
    "filePath": "path/to/${Stream}_${Sequence}_1.mp4",
    "infoPath": "path/to/file_1.xml",
    // "interval": 30000,
    //   "schedule" : "*/2 * *",
    "metadata": "dddd",
    "segmentationRule": "discontinuity"
}
```
For the macro used in filePath and infoPath, please refer to the link below.

[Rest API]
https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/recording

[Macro]
https://airensoft.gitbook.io/ovenmediaengine/recording#appendix-a.-macro-definition-for-the-recording-path
